controllers/operandrequest/reconcile_operator_test.go

```diff
@@ -326,4 +343,95 @@ func TestCheckSubAnnotationsForUninstall(t *testing.T) {
+			},
+			Labels: map[string]string{
+				constant.OpreqLabel: "true",
+			},
+		},
+	}
+
+	uninstallOperator, uninstallOperand = checkSubAnnotationsForUninstall(reqNameA, reqNsOld, opNameV3, operatorv1alpha1.InstallModeNamespace, sub)
+
+	assert.False(t, uninstallOperator)
+	assert.False(t, uninstallOperand)
+
+	assert.NotContains(t, sub.Annotations, reqNsOld+"."+reqNameA+"."+opNameV3+"/request")
+	assert.NotContains(t, sub.Annotations, reqNsOld+"."+reqNameA+"."+opNameV3+"/operatorNamespace")
+	assert.NotContains(t, sub.Annotations, reqNsOld+"."+reqNameA+"."+opNameV3+"/config")
+	assert.Contains(t, sub.Annotations, reqNsOld+"."+reqNameB+"."+opNameV4+"/request")
+	assert.Contains(t, sub.Annotations, reqNsOld+"."+reqNameB+"."+opNameV4+"/operatorNamespace")
+	assert.Contains(t, sub.Annotations, reqNsOld+"."+reqNameB+"."+opNameV4+"/config")
+
+	// Test case 10: Different config names - operands SHOULD be uninstalled
+	// When uninstalling one operator and other operators have different config names,
+	// the operands SHOULD be uninstalled
+	sub = &olmv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: reqNsOld,
+			Annotations: map[string]string{
+				reqNsOld + "." + reqNameA + "." + opNameV3 + "/request":           opChannelV3,
+				reqNsOld + "." + reqNameA + "." + opNameV3 + "/operatorNamespace": reqNsOld,
+				reqNsOld + "." + reqNameA + "." + opNameV3 + "/config":            opConfigV3,
+				reqNsOld + "." + reqNameB + "." + opNameV4 + "/request":           opChannelV4,
+				reqNsOld + "." + reqNameB + "." + opNameV4 + "/operatorNamespace": reqNsOld,
+				reqNsOld + "." + reqNameB + "." + opNameV4 + "/config":            opConfigV4,
+			},
+			Labels: map[string]string{
+				constant.OpreqLabel: "true",
+			},
+		},
+	}
+
+	uninstallOperator, uninstallOperand = checkSubAnnotationsForUninstall(reqNameA, reqNsOld, opNameV3, operatorv1alpha1.InstallModeNamespace, sub)
+
+	assert.False(t, uninstallOperator)
+	assert.True(t, uninstallOperand)
+
+	assert.NotContains(t, sub.Annotations, reqNsOld+"."+reqNameA+"."+opNameV3+"/request")
+	assert.NotContains(t, sub.Annotations, reqNsOld+"."+reqNameA+"."+opNameV3+"/operatorNamespace")
+	assert.NotContains(t, sub.Annotations, reqNsOld+"."+reqNameA+"."+opNameV3+"/config")
+	assert.Contains(t, sub.Annotations, reqNsOld+"."+reqNameB+"."+opNameV4+"/request")
+	assert.Contains(t, sub.Annotations, reqNsOld+"."+reqNameB+"."+opNameV4+"/operatorNamespace")
+	assert.Contains(t, sub.Annotations, reqNsOld+"."+reqNameB+"."+opNameV4+"/config")
+
+	// Test case 11: Version-suffixed operator names with shared config (the actual issue case)
+	// This simulates the exact issue seen in logs where operator names have version suffixes
+	sub = &olmv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "keycloak",
+			Annotations: map[string]string{
+				"keycloak.example-service.keycloak-operator/request":               "stable",
```

Comment:
Could we use variables? We could re-use variable `reqNsOld` for namespace and `reqNameA` OperandRequest name, and just introduce new variable for `keycloak-operator` and `keycloak-operator-v26`